### ISSUE TITLE
Generalize malign to Semigroup?

### ISSUE THREAD

**Issue by sjakobi** [isomorphism/these#69]
The type of `Data.Align.malign` could be generalized from 

```
(Align f, Monoid a) => f a -> f a -> f a
```

to

```
(Align f, Semigroup a) => f a -> f a -> f a
```

(This was pointed out by @enobayram on [r/haskell](https://www.reddit.com/r/haskell/comments/5hksnn/algebraic_blindness/db1tuly/))

A downside to this generalization would be the loss of the great mnemonic in the function name ("Monoid-align"). Maybe a renaming to "salign" should even be considered but I don't think that's worth it.

WRT dependencies, the generalization would mean that the previously transitive dependency on the `semigroups` package (via `semigroupoids`) would now turn into a direct dependency.

**Comment by sjakobi** [comment]
> I'd rather introduce new salign name

Would we then have `salign` in addition to `malign` or should it replace `malign`?

**Comment by phadej** [comment]
See #70

**Comment by enobayram** [comment]
Hi mates, I think we can save the mnemonic! malign only ever combines 2 things with a natural order, so its binary operation may not be associative. It might as well be just a magma! Though I guess that's a less common abstraction in Haskell, so I'm not sure it's a good idea.

**Comment by phadej** [comment]
Released in [`these-0.7.3`](http://hackage.haskell.org/package/these-0.7.3)
